cns/requestcontroller/example/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"time"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/requestcontroller"
+	"github.com/Azure/azure-container-networking/cns/requestcontroller/kubernetes"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+)
+
+func goRequestController(rc requestcontroller.RequestController) {
+	//Start the RequestController which starts the reconcile loop
+	if err := rc.StartRequestController(); err != nil {
+		logger.Errorf("Error starting requestController: %v", err)
+	}
+	// After calling StartRequestController, there needs to be some pause or, the calling program
+	// must keep running because the reconcile loop is spawned off on a different go-routine inside
```

Comment:
I a real world example we'd want soemthign that would block until a control c signal right?

Reply:
Okay this is doing that already. 
k8sRC.mgr.Start(ctrl.SetupSignalHandler()); err != nil {

Any other reason we'd send a cancel signal outside of termination? Do we want to wait for mgr to finish up before we exit (an exit channel might be useful?

Reply:
Adding an exitChan parameter to StartRequestController(exitChan), this exitChan will be notified when the mgr receives SIGINT or SIGTERM. let me know if that's what you mean (usage in example/main.go)